extra_data/run_files_map.py

```diff
@@ -121,13 +129,22 @@ def load(self):
                 st = os.stat(osp.join(self.directory, filename))
             except OSError:
                 continue
-            if (st.st_mtime == info['mtime']) and (st.st_size == info['size']):
+            if self._cache_info_valid(info, st):
                 self.files_data[filename] = info
 
         if loaded_data:
             dt = time.monotonic() - t0
             log.debug("Loaded cached files map in %.2g s", dt)
 
+    @staticmethod
+    def _cache_info_valid(info, file_stat: os.stat_result):
+        # Ignore the cached info if the file size or mtime have changed, or
+        # if it is missing expected keys (likely
+        added_keys = {'suspect_train_indices', 'legacy_sources'}
+        return ((file_stat.st_mtime == info['mtime'])
+                and (file_stat.st_size == info['size'])
+                and added_keys.issubset(info.keys()))
```

Comment:
Hmmm, I'm wondering if checking `cls.expected_cache_keys.issubset(info.keys())` wouldn't be simpler.

Reply:
It would, good idea.